diff --git a/arch/arm/src/stm32/stm32f10xxx_rcc.c b/arch/arm/src/stm32/stm32f10xxx_rcc.c
index bd89aa0e57..7cc5f9991e 100644
--- a/arch/arm/src/stm32/stm32f10xxx_rcc.c
+++ b/arch/arm/src/stm32/stm32f10xxx_rcc.c
@@ -524,7 +524,11 @@ static void stm32_stdclockconfig(void)
   /* Enable HSE */
 
   regval  = getreg32(STM32_RCC_CR);
+#ifdef STM32_HSEBYP_ENABLE          /* May be defined in board.h header file */
+  regval |= RCC_CR_HSEBYP;          /* Enable HSE clock bypass */
+#else
   regval &= ~RCC_CR_HSEBYP;         /* Disable HSE clock bypass */
+#endif
   regval |= RCC_CR_HSEON;           /* Enable HSE */
   putreg32(regval, STM32_RCC_CR);
 
@@ -640,7 +644,11 @@ static void stm32_stdclockconfig(void)
       /* Enable External High-Speed Clock (HSE) */
 
       regval  = getreg32(STM32_RCC_CR);
+#ifdef STM32_HSEBYP_ENABLE          /* May be defined in board.h header file */
+      regval |= RCC_CR_HSEBYP;          /* Enable HSE clock bypass */
+#else
       regval &= ~RCC_CR_HSEBYP;         /* Disable HSE clock bypass */
+#endif
       regval |= RCC_CR_HSEON;           /* Enable HSE */
       putreg32(regval, STM32_RCC_CR);
 
